include/exadg/postprocessor/solution_field.h

```diff
@@ -35,23 +35,103 @@ enum class SolutionFieldType
 };
 
 template<int dim, typename Number>
-class SolutionField
+class SolutionField : public dealii::Subscriptor
 {
 public:
+  using VectorType = dealii::LinearAlgebra::distributed::Vector<Number>;
+
   SolutionField()
-    : type(SolutionFieldType::scalar), name("solution"), dof_handler(nullptr), vector(nullptr)
+    : recompute_solution_field([](VectorType &, VectorType const &, bool const) {}),
+      type(SolutionFieldType::scalar),
+      name("solution"),
+      dof_handler(nullptr),
+      is_available(false),
+      solution(nullptr)
+  {
+  }
+
+  void
+  reinit(VectorType const & solution_in)
+  {
+    is_available = false;
+    solution     = &solution_in;
+  }
+
+  void
+  evaluate(bool const unsteady)
```

Comment:
I would like to get rid of the parameter `unsteady`, which appears as function argument in four functions in this PR, but it seems as if it is only required by the assert in `compute_mean_velocity()`. I think we should find a better solution for this. What happens if we only create and access the mean velocity object in case of unsteady problems? i.e. `Postprocessor` has the notion of steady/unsteady problems and can care about this, but a solution field should not care about this.

Reply:
Only creating in the unsteady case is not as easy as it seems, since inside the relevant function it is not known if the simulation is unsteady. 
Instead, I moved the Assert to the point before mean_velocity is evaluated. This simplifies the code a lot, but comes at the price that the user has to think about checking if the simulation is unsteady before evaluating mean_velocity.

Reply:
I think this is exactly how it should look like. I like the current version.